### ISSUE TITLE
[Infrastructure] Deploy DevNet(s) of V1 

### ISSUE THREAD

**Issue by okdas** [pokt-network/pocket#134]
# Objective
Deploy the V1 LocalNet to a remote environment (e.g. AWS, GCP, etc…)

# Origin Document
No origin document necessary. A functional DevNet is the first step needed for a functional TestNet

# Goals / Deliverables
- V1 Configurations
    - [ ]  Generalize the V1 LocalNet infrastructure to run any number of validators
    - [ ]  Improve the V1 codebase to more easily configure nodes
        - [ ]  Genesis config
        - [ ]  Node config
- V1 Deployment
    - [ ]  Enable deploying multiple validators on a single machine
    - [ ]  Enable deploying multiple on different machines in different regions
    - [ ]  Enable deploying several different DevNets with different configs in parallel (i.e. personal/namespaced DevNets)
    - [ ]  Automatically deploy the HEAD on master with a preconfigured state
- V1 Visibility
    - [ ]  Gain visibility (e.g. dashboards, 3rd party tools, etc…) into all the DevNets deployed (configuration, status, distribution, sha, etc…)
    - [ ]  Provide the community with insight into where/when their contributions are deployed
- [ ]  Identify follow up and extensions vis TODOs/issues/milestones

## General milestone checklist
- [ ] Update all the relevant CHANGELOGs
- [ ] Update all the relevant READMEs
- [ ] Update the source code tree explanation
- [ ] Add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)
- [ ] Create followup milestones + issues
- [ ] Document small TODO along the way

# Maybe-goals
- [] Support starting a validator from a previous state (if not, then from genesis)
- 
# Non-goals
- DevNet wars: different permutations of different binaries
- Add support for other types of node roles (e.g. full nodes, service nodes, etc…)
- Dynamic addition/removal of validators from consensus

---

Creator: @okdas @Olshansk 
Co-Owners: TBD

**Comment by Olshansk** [comment]
Just jotting some ideas down, but we'll need a more proper design later.

Each user / owner / deployer of a DevNet should be able to do something like:

```bash
$ pocket_v1_cli devnet connect --host:port // connects to a devnet like similar to a k8s cluster
$ pocket_v1_cli devnet list // lists all the different types of nodes deployed on this devnet
$ pocket_v1_cli devnet cli --node=val1 <some_rpc_command> // trigger some call
$ pocket_v1_cli devnet cli --node=val1 ssh` // ssh into the node
```

cc @andrewnguyen22 and @adshmh for context

**Comment by okdas** [comment]
We decomposed this ticket into smaller github issues.
